app_dart/test/request_handlers/update_task_status_test.dart

```diff
@@ -120,5 +120,31 @@ void main() {
       expect(task.status, 'Failed');
       expect(task.attempts, 1);
     });
+
+    test('non-task key requests can update tasks', () async {
```

Comment:
one more test for cases where no task is found?

Reply:
Good catch, added.